apps/auction2/reducers.js

```diff
@@ -0,0 +1,29 @@
+import { combineReducers } from 'redux'
+
+const initialState = {
+  artworks: [],
+  listView: false
+}
+
+function auctionArtworks(state = initialState, action) {
+  switch (action.type) {
+  case 'UPDATE_ARTWORKS':
+    return {
+      ...state,
+      artworks: action.artworks
+    }
+  case 'TOGGLE_LIST_VIEW':
+    return {
+      ...state,
+      listView: action.listView
+    }
```

Comment:
This is like not required -- but i think it would be really good since you're laying down patterns for going forward -- is to use https://github.com/substantial/updeep in the reducer for updating state. Although the library can do a lot, for most use-cases its actually really simple — when you use this syntax:

```javascript
return { 
  …state
  listView: action.listView
}
```

the spread operator `...state` only makes a shallow copy, and `listView` only goes one level deep, so we're all good. BUT, If you're operating on a deeply-nested object, its difficult to update elegantly because you have to ensure that the deeply nested branch is cloned completely since Redux expects an entirely new object. Updeep solves this (as do other libraries such as baobab, Immutable.js, etc), but the difference between them all is that Updeep operates on regular objects and so there's no translation between things and you can treat it like normal JS. 

So, here's typical updeep use in practice:

```javascript
import u from 'updeep'

function auctionArtworks(state = initialState, action) {
  switch (action.type) {
  case 'UPDATE_ARTWORKS':
    return u({
      artworks: action.artworks
    }, state)
  case 'TOGGLE_LIST_VIEW':
    return u({
      listView: action.listView
    }, state)
  case 'SOME_DEEPLY_NESTED_PROP':
    return u({
      some: { 
        deeply: {
          nested: {
            prop: action.heynow
          }
        }
      }
    }, state)
  default:
    return state
  }
}
```

At some point every redux dev runs into this nesting problem, so it would probably be good to solve it right at the gate.